### PR TITLE
Evolution etesync

### PR DESCRIPTION
**Pull request by Pacman99**
###### Motivation for this change
Use etesync in gnome by adding the evolution plugin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
- [x] Tested etesync integration with services.gnome.evolution-data-server.plugins and programs.evolution.plugins

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130686` at f9c62b03 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>evolution-etesync</li>
    <li>libetebase</li>
  </ul>
</details>

File: pkgs/development/libraries/libetebase/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  '';
+
+  meta = with lib; {
+    description = "A C library for Etebase";
+    homepage = "https://github.com/etesync/libetebase/";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ pacman99 ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libetebase/default.nix:59]:
```suggestion
  };
}
```

File: pkgs/development/libraries/libetebase/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    runHook preInstall
+    mkdir -p $out
+    make install PREFIX=$out
+
+    # For references in evolution-etesync
+    ln -s $out/include/etebase/etebase.h $out/include/etebase.h
+    ln -s $out/lib/libetebase.so $out/lib/libetebase.so.0
+
+    runHook postInstall
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libetebase/default.nix:50]:
```suggestion
  installFlags = [ "PREFIX=$out" ];

  postInstall = ''
    # For references in evolution-etesync
    ln -s $out/include/etebase/etebase.h $out/include/etebase.h
    ln -s $out/lib/libetebase.so $out/lib/libetebase.so.0
  '';
```

File: pkgs/development/libraries/libetebase/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+, fetchFromGitHub
+}:
+stdenv.mkDerivation rec {
+  pname = "libetebase";
+  version = "v0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "etesync";
+    repo = "libetebase";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libetebase/default.nix:15]:
```suggestion
  version = "0.5.0";

  src = fetchFromGitHub {
    owner = "etesync";
    repo = "libetebase";
    rev = "v${version}";
```

File: pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   passthru = {
+    updateScript = gnome.updateScript {
+      packageName = "evolution-etesync";
+    };
+  };
+
+  meta = with lib; {
+    description = "EteSync plugin for Evolution";
+    homepage = "https://gitlab.gnome.org/GNOME/evolution-etesync";
+    license = "LGPL-2.1-only";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix:65]:
```suggestion
    license = license.lgpl2Only;
```

File: pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  cmakeFlags = [
+    # don't try to install into ${evolution}
+    "-DFORCE_INSTALL_PREFIX=ON"
+  ];
+
+   passthru = {
+    updateScript = gnome.updateScript {
+      packageName = "evolution-etesync";
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix:60]:
```suggestion
  passthru = {
    updateScript = gnome.updateScript {
      packageName = "evolution-etesync";
    };
  };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg build libetebase evolution-etesync

File: pkgs/development/libraries/libetebase/default.nix [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    owner = "etesync";
+    repo = "libetebase";
+    rev = "v${version}";
+    sha256 = "0r6404lgaawwkEsF3HNB+eSJ+nkEeUHUiIKUNxFDAGc=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "sha256-NN7ouGCfVYyIwTKezROWirmdySaCitbLoIN+FdeSZUE=";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libetebase/default.nix:22]:
```suggestion
    sha256 = "0w04n78b9vjsd1vaz1yrqwr3s7a38qag09qsqc71rvbsar2rmzwz";
```

File: pkgs/development/libraries/libetebase/default.nix [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+  checkInputs = with rustPlatform; [
+    cargoCheckHook
+  ];
+
+  installFlags = [ "PREFIX=$out" ];
+
+  postInstall = ''
+    # For references in evolution-etesync
+    ln -s $out/include/etebase/etebase.h $out/include/etebase.h
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libetebase/default.nix:44]:
```
ln: failed to create symbolic link '/nix/store/98mfn51flkabvhgyx4jph453adpag5mi-libetebase-0.5.0/include/etebase.h': No such file or directory
```

File: pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  buildInputs = [
+    evolution-data-server
+    evolution
+    sqlite
+    libgdata
+    gtk3
+    webkitgtk
+    libmspack
+    libetebase
+    autoPatchelfHook
```
Comment (jtojnar) [inline pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix:48]: This should go into `nativeBuildInputs`.

File: pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  passthru = {
+    updateScript = gnome.updateScript {
+      packageName = "evolution-etesync";
+    };
+  };
+
+  meta = with lib; {
+    description = "EteSync plugin for Evolution";
+    homepage = "https://gitlab.gnome.org/GNOME/evolution-etesync";
+    license = licenses.lgpl2Only;
```
Comment (jtojnar) [inline pkgs/applications/networking/mailreaders/evolution/evolution-etesync/default.nix:65]: Why not LGPL-2.1-or-later? That seems to be listed in the random files I checked.

**Comment by Pacman99** [comment]
Completely forgot about this PR, I just updated to apply the suggestions, thanks for the review!

File: pkgs/development/libraries/libetebase/default.nix [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+  buildInputs = [
+    openssl
+  ];
+
+  checkInputs = with rustPlatform; [
+    cargoCheckHook
+  ];
+
+  installFlags = [ "PREFIX=$out" ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libetebase/default.nix:40]: You cannot use shell variable syntax in `installFlags`, make will not be able to interpolate them. You either need to use Makefile syntax `$(out)`, or a Nix placeholder:

```patch
--- a/pkgs/development/libraries/libetebase/default.nix
+++ b/pkgs/development/libraries/libetebase/default.nix
@@ -37,14 +37,9 @@ stdenv.mkDerivation rec {
     cargoCheckHook
   ];
 
-  installFlags = [ "PREFIX=$out" ];
-
-  postInstall = ''
-    # For references in evolution-etesync
-    mkdir -p $out/include
-    ln -s $out/include/etebase/etebase.h $out/include/etebase.h
-    ln -s $out/lib/libetebase.so $out/lib/libetebase.so.0
-  '';
+  installFlags = [
+    "PREFIX=${placeholder "out"}"
+  ];
 
   meta = with lib; {
     description = "A C library for Etebase";
```

File: pkgs/development/libraries/libetebase/default.nix [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+  postInstall = ''
+    # For references in evolution-etesync
+    mkdir -p $out/include
+    ln -s $out/include/etebase/etebase.h $out/include/etebase.h
+    ln -s $out/lib/libetebase.so $out/lib/libetebase.so.0
```
Comment (jtojnar) [inline pkgs/development/libraries/libetebase/default.nix:46]: This is upstream bug, they [set the soname](https://github.com/etesync/libetebase/blob/e06fe2461249b7e2f535fc485bf9a1abd67feff9/build.rs#L13) but not the symlink. Could you open a PR upstream to do that in the [Makefile](https://github.com/etesync/libetebase/blob/master/Makefile#L42)?

**Pull request closed without merging**